pyaedt/edb_core/edb_data/sources.py

```diff
@@ -274,34 +275,53 @@ def net(self, value):
     def net_name(self):
         return self._edb_pin_group.GetNet().GetName()
 
+    @property
+    def terminal(self):
+        """Terminal."""
+        from pyaedt.edb_core.edb_data.terminals import PinGroupTerminal
+
+        term = PinGroupTerminal(self._pedb, self._edb_pin_group.GetPinGroupTerminal())
+        return term if not term.is_null else None
+
     @pyaedt_function_handler()
-    def _create_pin_group_terminal(self, is_reference=False):
-        pg_term = self._edb_pin_group.GetPinGroupTerminal()
-        pin_group_net = self._edb_pin_group.GetNet()
-        if pin_group_net.IsNull():  # pragma: no cover
-            pin_group_net = list(self._edb_pin_group.GetPins())[0].GetNet()
-        if pg_term.IsNull():
-            return self._pedb.edb_api.cell.terminal.PinGroupTerminal.Create(
-                self._active_layout,
-                pin_group_net,
-                self.name,
-                self._edb_pin_group,
-                is_reference,
-            )
-        else:
+    def _create_terminal(self, name=None):
+        """Create a terminal on the pin group.
+
+        Parameters
+        ----------
+        name : str, optional
+            Name of the terminal. The default is ``None``, in which case a name is
+            automatically assigned.
+
+        Returns
+        -------
+        :class:`pyaedt.edb_core.edb_data.terminals.PinGroupTerminal`
+        """
+        pg_term = self.terminal
+        if not name:
+            name = self.name
+
+        if pg_term:
             return pg_term
+        else:
+            from pyaedt.edb_core.edb_data.terminals import PinGroupTerminal
+
+            term = PinGroupTerminal(self._pedb)
+
+            term = term.create(name, self.net_name, self.name)
```

Comment:
I don't quite understand. the method _create_terminal creates actually a PingroupTerminal whichius different. I found it confusing

Reply:
ok with this one LGTM